pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
We can't error on differences between the live state and the patch. The live state includes, for instance, default values. If the user didn't previously configure the fields, they didn't care what the values were. If they are now configuring those fields (i.e., in the patch), now they care, and should set them, regardless of their current values in the live state.